### PR TITLE
refactor: RHICOMPL-1474 inventory consumer incl. report download and validation

### PR DESCRIPTION
**Pull request by vkrizan**
Refactoring to plug in auditing within Kafka inventory consumer.

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [X] Input Validation
- [X] Output Encoding
- [X] Authentication and Password Management
- [X] Session Management
- [X] Access Control
- [X] Cryptographic Practices
- [X] Error Handling and Logging
- [X] Data Protection
- [X] Communication Security
- [X] System Configuration
- [X] Database Security
- [X] File Management
- [X] Memory Management
- [X] General Coding Practices

### CODE REVIEW

**Review by skateman — APPROVED** [review]
Not sure how much my word weighs on my second day in this team, but the code makes sense to me :+1:

**Comment by akofink** [comment]
I'm seeing some issues with this PR checked out when trying to parse reports. I'm still trying to track down what exactly breaks:

```
==> log/inventory-consumer.log <==
E, [2021-03-02T15:23:01.016811 #1] ERROR -- : Failed to process platform.inventory.events/0 at 18..18: closed stream
W, [2021-03-02T15:23:01.016906 #1]  WARN -- : Pausing partition platform.inventory.events/0 for 10.0 seconds
```

File: app/jobs/delete_host.rb [hunk 1]
```diff
@@ -4,12 +4,23 @@
...
+
+  private
+
+  def remove_related(host_id)
+    num_removed = 0
     Host.transaction do
-      Sidekiq.logger.info("Deleting related records for host #{message['id']}")
-      [RuleResult, TestResult, PolicyHost].each do |model|
-        model.where(host_id: message['id']).delete_all
+      Sidekiq.logger.info("Deleting related records for host #{host_id}")
```
Comment (akofink) [inline app/jobs/delete_host.rb:19]: Should this be audit logged? Why count `num_removed` if it's not used (am I missing something)?

Comment (vkrizan) [reply app/jobs/delete_host.rb:19]: See #767

Comment (akofink) [reply app/jobs/delete_host.rb:19]: Oh! right. :laughing:

**Comment by vkrizan** [comment]
@akofink Fixed by replacing the lazy enumeration with just a filename filler.

File: app/services/reports_tar_reader.rb [hunk 2]
```diff
@@ -10,15 +10,22 @@ def initialize(file)
   end
 
   def reports
-    tar_extract = Gem::Package::TarReader.new(Zlib::GzipReader.new(@file))
-    tar_extract.rewind # The extract has to be rewinded after every iteration
-    tar_extract.map do |file|
-      file.read if file.header.name.match(/oscap_results/)
+    extract.map do |file|
+      filename = File.basename(file.header.name)
```
Comment (vkrizan) [inline app/services/reports_tar_reader.rb:14]: this one seems to be truncated :frowning_face:

**Review by akofink — APPROVED** [review]
Thanks, @vkrizan! Seems to continue to work as expected.

File: app/services/reports_tar_reader.rb (hunk 2 above)
Comment (vkrizan) [reply app/services/reports_tar_reader.rb:14]:
```
struct tarfile_entry_posix {
  char name[100];     # ASCII + (Z unless filled)
            ^^^
```

Comment (vkrizan) [reply app/services/reports_tar_reader.rb]: Okay, so I've dropped the filename gathering and rather changed it to take profile id from parsed content.

**Review by skateman — APPROVED** [review]
The Seal of Approval

**Pull request merged**